0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md

```diff
@@ -1,47 +1,104 @@
+2. The receiver endpoint responds with the receiver info, including the recipient's ILP address.
+3. The sender choses an amount of their currency or assets to send.
+4. The sender's SPSP client uses its ILP module to quote how much of the recipient's currency or asset type will be delivered to the recipient's ILP address for the given source amount.
+5. The sender accepts the quote.
+6. The sender's SPSP client submits the payment information, including the destination amount, to the receiver endpoint.
+7. The receiver endpoint responds with an ILP Packet and Crypto Condition corresponding to the destination amount.
+8. The sender's SPSP client uses its ILP module to create a transfer **using the chosen source amount, NOT by quoting the ILP packet**, and attach the ILP packet to the transfer.
+9. The receiver's ILP module registers the incoming transfer held pending the fulfillment of the Crypto Condition. It validates that the transfer matches the packet and regenerates the condition fulfillment using the [Interactive Transport Protocol (ITP)](../0011-interactive-transport-protocol). The ILP module submits the fulfillment to execute the transfer and claim the funds.
+10. The sender's SPSP client receives a notification from its ILP module that the transfer has been executed, including the condition fulfillment from the recipient, and notifies the sender that the payment is completed.
+
+#### Invoice
+
+1. The sender's SPSP client queries the receiver endpoint.
+2. The receiver endpoint responds with the receiver info, including the invoice status, amount, and currency code.
+3. The sender's SPSP client submits the sender's info, including the sender's ILP address to the receiver endpoint.
+4. The receiver endpoint responds with an ILP Packet and Crypto Condition corresponding to the destination amount.
+5. The sender's SPSP client uses its ILP module to get a quote in their currency or asset type for the ILP transfer.
+6. The sender accepts the quote.
+7. The sender's SPSP client uses its ILP module to initiate the ILP transfer.
+8. The receiver's ILP module registers the incoming transfer held pending the fulfillment of the Crypto Condition. It validates that the transfer matches the packet and regenerates the condition fulfillment using the [Interactive Transport Protocol (ITP)](../0011-interactive-transport-protocol). The ILP module submits the fulfillment to execute the transfer and claim the funds.
+9. The sender's SPSP client receives a notification from its ILP module that the transfer has been executed, including the condition fulfillment from the recipient, and notifies the sender that the payment is completed.
+
+## Specification
+
+The receiver endpoint will respond to HTTP `GET` and `POST` requests in the following manner:
+
+### Query (`GET <receiver>`)
+
+The sender queries the receiver endpoint to get information about the type of payment that can be made to this receiver:
 
 ``` http
 GET /api/receivers/bob HTTP/1.1
```

Comment:
I think there's an argument to be made for returning all the data we think (or say, since we're defining the scope of this protocol) senders will need in the Query response. Who the receiver is is probably included in that. Are what accounts the receiver has relevant? Arguably, they get to decide which account you're dealing with.

There are three ways I could see us going regarding query parameters:

1. The sender must treat the receiver endpoint as completely opaque. The URL MAY include query parameters but those are decided by the recipient and cannot be modified by the sender.
2. The recipient can include some query parameters in the URL but we also define some standard ones that receivers SHOULD understand.
3. The base receiver URL cannot have query parameters in it and we standardize some query parameters that the receiver SHOULD understand.